### ISSUE TITLE
Track total data $ volume consumed / day, and total data $ volume bought / day

### ISSUE THREAD

**Issue by trentmc** [oceanprotocol/market#380]
Motivations:
* these are among our most important KPIs, to guide our actions on traction, and for external comms. That's why priority = high.
* data $ volume consumed is the best proxy for usefulness of data. It's a proxy for the value created by Ocean
* data $ volume bought is a proxy for the overall size of the ocean data economy, including speculation (which is an important factor)

What:
* show "total data $ volume consumed / day", and "total data $ volume bought / day" in the footer of market.oceanprotocol.com, in the same region as TVL

Notes: 
* to implement, can use a similar pattern to TVL. The subgraphs calculate in terms of OCEAN, then do a final conversion to EUR or USD.
* there's already another issue for $ volume at a pool level.

**Comment by trentmc** [comment]
cc @kremalicious

**Comment by kremalicious** [comment]
blocked by https://github.com/oceanprotocol/ocean-subgraph/issues/10

**Comment by mihaisc** [comment]
most of these will be solvable after https://github.com/oceanprotocol/ocean-subgraph/issues/180 . For consume volume i think we can do a sum of orders like: spotPrice at consume timestamp 1 + spotPrice at consume timestamp 2 , and have this per day. The issue is that the spotPrice is expressed in the pair token of that pool ( if pool is dt/ocean then spotPrice is in ocean, if pool is dt/eth spotPrice is in eth ), thus we have an array of tokens. Another step would be to convert all of then in fiat at the conversion rate of that day and i believe this would be the consume price of day x.

**Comment by mihaisc** [comment]
Related to https://github.com/oceanprotocol/market/issues/871

**Comment by trentmc** [comment]
FYI we've built what we needed for this, for DF. Maybe some of it's useful for this issue. Or maybe this issue is sufficiently redundant with the work done below and outstanding issue below that we can now close this one.

Key things that DF has:

- [query.py::getDTVolumes()](https://github.com/oceanprotocol/df-py/blob/35f989c658825ec379fc1a55fcf02410710e2c6e/util/query.py#L202) - gives volume for a datatoken (DT) in a block range.
- [query.py::getPoolVolumes()](https://github.com/oceanprotocol/df-py/blob/35f989c658825ec379fc1a55fcf02410710e2c6e/util/query.py#L174) gives volume for a pool in a block range, using getDTVolumes().
- Since the above operate on block numbers, [blocktime.py](https://github.com/oceanprotocol/df-py/blob/main/util/blocktime.py) is a library to convert from a specified day/time to block number.

Then, DF has an infrastructure around that to meet the needs of DF:
- the above can be called via CLI or via library
- it creates useful CSVs
- which are consumed for weekly DF rewards payout
- and which are consumed realtime in the DF webapp (df-web repo)

[Here's](https://github.com/oceanprotocol/df-py/blob/main/images/data-flow.png) how the data flows. [GSlide source](https://docs.google.com/presentation/d/11Nu_emSgw3iYZKDcmnvZC_aOM00MDyYTtTWQP4PYox4/edit#slide=id.p)

One thing that DF does _not_ have yet, but is slated for later, and would be useful to the broader Ocean ecosystem, is a Dune (or Dune-style) Dashboard to track TVL, DCV, etc over time. This is in [df-py#19](https://github.com/oceanprotocol/df-py/issues/19)

**Comment by mihaisc** [comment]
Cool, thanks, i'll spec something lighter then and maybe think of some integration/link to df. No need to do redundant stuff
